test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,651 @@
+        if input_type in {torch.Tensor, PIL.Image.Image, datapoints.Image}:
+            input = make_image(size=self.INPUT_SIZE, dtype=dtype or torch.uint8, device=device)
+            if input_type is torch.Tensor:
+                input = input.as_subclass(torch.Tensor)
+            elif input_type is PIL.Image.Image:
+                input = F.to_image_pil(input)
+        elif input_type is datapoints.BoundingBox:
+            input = make_bounding_box(
+                format=datapoints.BoundingBoxFormat.XYXY,
+                spatial_size=self.INPUT_SIZE,
+                dtype=dtype or torch.float32,
+                device=device,
+            )
+        elif input_type is datapoints.Mask:
+            input = make_segmentation_mask(size=self.INPUT_SIZE, dtype=dtype or torch.uint8, device=device)
+        elif input_type is datapoints.Video:
+            input = make_video(size=self.INPUT_SIZE, dtype=dtype or torch.uint8, device=device)
+
+        return input
+
+    def _check_size(self, input, output, *, size, max_size):
+        if isinstance(size, int) or len(size) == 1:
+            if not isinstance(size, int):
+                size = size[0]
+
+            old_height, old_width = F.get_spatial_size(input)
+            ratio = old_width / old_height
+            if ratio > 1:
+                new_height = size
+                new_width = int(ratio * new_height)
+            else:
+                new_width = size
+                new_height = int(new_width / ratio)
+
+            if max_size is not None and max(new_height, new_width) > max_size:
+                # Need to recompute the aspect ratio, since it might have changed due to rounding
+                ratio = new_width / new_height
+                if ratio > 1:
+                    new_width = max_size
+                    new_height = int(new_width / ratio)
+                else:
+                    new_height = max_size
+                    new_width = int(new_height * ratio)
+
+        else:
+            new_height, new_width = size
+
+        assert F.get_spatial_size(output) == [new_height, new_width]
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("interpolation", INTERPOLATION_MODES)
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_image_tensor(self, size, interpolation, use_max_size, antialias, dtype, device):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        if device == "cuda":
+            if dtype.is_floating_point:
+                check_cuda_vs_cpu_atol = 1 / 255
+            elif interpolation is transforms.InterpolationMode.BICUBIC:
+                check_cuda_vs_cpu_atol = 30
```

Comment:
We have a similar high value in v1

https://github.com/pytorch/vision/blob/b9a1984cc98fd72b93ab340e4bdd9c33bdbe326b/test/test_functional_tensor.py#L589-L595

Instead of setting a high value here and potentially risking a flaky test, we could also simply disable the check for this setting, i.e. setting `check_cuda_vs_cpu=False`. I don't think we will catch any bug here with a tolerance as high as this.

Reply:
Thinking about this more, IMO disabling is the right choice here. I went for that in a83f9fbc597e307345a7e824acae7bf781e79181. LMK if you prefer high tolerance.

Reply:
Can't hurt to keep it IMHO. It's a reminder that CUDA doesn't support uint8 for bicubic and that's why there's a non-trivial difference here: CUDA uses the float path (even if the input is uint8) while on CPU we natively support uint8, and the 2 paths have different implementations